WG_DYNAMIC_DATA_AND_CAPABILITIES.md

```diff
@@ -0,0 +1,224 @@
+*   **Remote Presence**: allow collaborating peers to be informed about each others' presence and actions
+*   **Authentication**: allow a peer to authenticate a given piece of data
+*   **Identity-based P2P secret sharing**: allow a peer to share a secret that will be authenticated and secret
+*   **Personal** **feeds**: allow any peer to expose a set of activity feeds to any permissioned peer
+*   **Causal broadcast channel**: create a group messaging channel that respects causality
+*   **Collaborative replication**: multi-writer-enabled data types
+    *   Op-based CRDTs (Conflict-free replicated data types that are based on a growing set of operations)
+    *   Delta state-based CRDTs Conflict-free replicated data types that are based on transmitting deltas or entire states)
+
+#### 1.2. (DX) Peer-base release management is well defined and optimal (P0)
+
+*   Change logs
+*   Support migrations for breaking changes
+*   Long-lived schema management
+
+#### 1.3. (DX) A methodology for supporting peer-base libraries is defined and implemented (P0)
+
+(look at other OSS projects like IPFS, Node.js)
+
+#### 1.4. (DX) Developers building dynamic DApps on top of IPFS / libp2p are supported (P1)
+
+Identify package maintainers and their responsibilities.
+
+#### 1.5. (DX) Public runtime infrastructure (run by PL) exists and is stable (P1)
+
+Services:
+*   Pinners
+*   Gateways
+*   Websocket-star
+*   Replace websocket-star with:
+    *   Rendez-vous nodes
+    *   Circuit-relay nodes
+*   Explore:
+    *   Signaling (for signal persistence and execution (support for offline delivery, push messages, etc.))
+
+Notes:
+*   Make sure we outsource it to the infra team.
+*   Define QoS contract.
+*   Gather data about infrastructure usage.
+*   Private infrastructure: maybe start conversation about enabling this (talk about features, economics, …).
+*   (Talk to partners, like Textile, about needs, scaling, volunteering, ...)
+
+#### 1.6. (DX) Deployment tools, code, instructions and images exist, and are supported (P2)
+
+For deploying apps and all the required services.
+
+Notes:
+*   (Requires cooperation with IPFS core, infra, …?)
+*   Define which tools should exist
+*   Assess which tools exist, who maintains them
+*   Idea: piggyback on IPFS desktop?
+*   Idea: support infrastructure donation?
+
+#### 1.7. (DX) peer-base v1 has been released (P3)
+
+Depends on M1.1.
+Signifies stability to developers.
+
+What does it contain, look like?
+
+#### 1.8. (DX) Online Materials (text mainly, perhaps video) exist and are supported (P3)
+
+*   Logo and Website
+*   Text tutorials first (easier to maintain)
+    *   Beginner
+    *   Intermediate
+    *   Advanced
+*   Video perhaps, but only for events
+
+#### 1.9. (DX) The working group has partnered with a third party on development of a production-quality collaborative application (P4)
```

Comment:
@pgte @daviddias Should this still be P4 if we're planning to focus more on developers than end users?

Reply:
@aschmahmann good call, bumped this to P1.